pkg/urbit/vere/io/ames.c

```diff
@@ -445,39 +686,164 @@ _ames_recv_cb(uv_udp_t*        wax_u,
               unsigned         flg_i)
 {
   u3_ames* sam_u = wax_u->data;
+  c3_o     pas_o = c3y;
+  c3_y*    byt_y = (c3_y*)buf_u->base;
+  c3_y*    bod_y = byt_y + 4;
+  u3_head  hed_u;
+
+  //  ensure a sane message size
+  //
+  if (4 >= nrd_i) {
+    pas_o = c3n;
+  }
+  //  unpack the packet header
+  //
+  else {
+    c3_w hed_w = (byt_y[0] <<  0)
+               | (byt_y[1] <<  8)
+               | (byt_y[2] << 16)
+               | (byt_y[3] << 24);
+
+    hed_u.ver_y = hed_w & 0x7;
+    hed_u.mug_l = (hed_w >> 3) & ((1 << 20) - 1);
```

Comment:
The mask should just be `0xfffff`, maybe with `((1 << 20) - 1)` in a comment if you'd like.